hyprtester/src/tests/main/snap.cpp

```diff
@@ -0,0 +1,174 @@
+        expectSnapMove({x, x}, &(predict = {GAP, GAP}));
+
+        x = MONITORGAP + END;
+        expectSnapMove({1920 - x, 1080 - x}, nullptr);
+        x -= 1;
+        expectSnapMove({1920 - x, 1080 - x}, &(predict = {1920 - END, 1080 - END}));
+
+        // test reserved area
+        const double RESERVED = 200;
+        const double RGAP     = RESERVED + RESP + BORDERSIZE;
+        const double REND     = RGAP + WINDOWSIZE;
+
+        x = MONITORGAP + RGAP;
+        expectSnapMove({x, x}, nullptr);
+        x -= 1;
+        expectSnapMove({x, x}, &(predict = {RGAP, RGAP}));
+
+        x = MONITORGAP + REND;
+        expectSnapMove({1920 - x, 1080 - x}, nullptr);
+        x -= 1;
+        expectSnapMove({1920 - x, 1080 - x}, &(predict = {1920 - REND, 1080 - REND}));
+    }
+}
+
+static bool test() {
+    NLog::log("{}Testing snap", Colors::GREEN);
+
+    // move to monitor HEADLESS-2
+    NLog::log("{}Moving to monitor HEADLESS-2", Colors::YELLOW);
+    OK(getFromSocket("/dispatch focusmonitor HEADLESS-2"));
+    NLog::log("{}Adding reserved monitor area to HEADLESS-2", Colors::YELLOW);
+    OK(getFromSocket("/keyword monitor HEADLESS-2,addreserved,200,200,200,200"));
+
+    // test on workspace "snap"
+    NLog::log("{}Dispatching workspace `snap`", Colors::YELLOW);
+    OK(getFromSocket("/dispatch workspace name:snap"));
+
+    // spawn a kitty terminal and move to (500,500)
+    NLog::log("{}Spawning kittyProcA", Colors::YELLOW);
+    if (!spawnFloatingKitty())
+        return false;
+
+    NLog::log("{}Expecting 1 window", Colors::YELLOW);
+    EXPECT(Tests::windowCount(), 1);
+
+    NLog::log("{}Move the kitty window to (500,500)", Colors::YELLOW);
+    OK(getFromSocket("/dispatch moveactive exact 500 500"));
+
+    // spawn a second kitty terminal
+    NLog::log("{}Spawning kittyProcB", Colors::YELLOW);
+    if (!spawnFloatingKitty())
+        return false;
+
+    NLog::log("{}Expecting 2 windows", Colors::YELLOW);
+    EXPECT(Tests::windowCount(), 2);
+
+    NLog::log("");
+    testSnap(false, false);
+
+    NLog::log("\n{}Turning on border_overlap", Colors::YELLOW);
+    OK(getFromSocket("/keyword general:snap:border_overlap true"));
+    testSnap(true, false);
+
+    NLog::log("\n{}Turning on respect_gaps", Colors::YELLOW);
+    OK(getFromSocket("/keyword general:snap:border_overlap false"));
+    OK(getFromSocket("/keyword general:snap:respect_gaps true"));
+    testSnap(false, true);
+
+    NLog::log("\n{}Turning on both border_overlap and respect_gaps", Colors::YELLOW);
+    OK(getFromSocket("/keyword general:snap:border_overlap true"));
+    testSnap(true, true);
+
+    // kill all
+    NLog::log("\n{}Killing all windows", Colors::YELLOW);
+    Tests::killAllWindows();
+
+    NLog::log("{}Expecting 0 windows", Colors::YELLOW);
+    EXPECT(Tests::windowCount(), 0);
+
+    NLog::log("{}Removing reserved monitor area from HEADLESS-2", Colors::YELLOW);
+    OK(getFromSocket("/keyword monitor HEADLESS-2,addreserved,0,0,0,0"));
```

Comment:
I'd just issue `hyprctl reload` instead over the socket, aka `/reload` to clear any unwanted dynamic stuff.